src/main/java/org/parabot/core/Context.java

```diff
@@ -85,7 +85,13 @@ public static double getJavaVersion() {
      * @return
      */
     public static Context getInstance(ServerProvider serverProvider) {
-        return instance == null ? instance = new Context(serverProvider) : instance;
+        if(instance == null)
```

Comment:
Could you elaborate on this choice of instance creation?

Reply:
If Context.getInstance() is called before you load a server. (Which it is if you open the script selector for instance) before you load the server, it will result in the Context instance to be set without the serverprovider ever being set. 

So this checks if the ServerProvider was set when we called Context.getInstance(serverProvider). If for some reason it wasn’t, it sets it.